R/mergeSEs.R

```diff
@@ -963,6 +960,36 @@ setMethod("right_join", signature = c(x = "ANY"),
     # Merge rowdata
     rd <- .join_two_tables(rd1, rd2, join)
     
+    # There might be duplicated rownames. This might occur when there are features
+    # with equal taxonomy data but merged datasets have some additional info
+    # that do not match with each other. --> collapse duplicated rows/features
+    # into one row.
+    dupl_rows <- rownames(rd)[ duplicated(rownames(rd)) ]
+    if( length(dupl_rows) > 0 ){
+        for( r in dupl_rows ){
+            # Get duplicated rows
+            temp <- rd[rownames(rd) %in% r, , drop = FALSE]
+            # Get uniwu rows
+            temp1 <- temp[1,]
+            temp2 <- temp[2,]
+            temp1 <- temp1[, !apply(temp1, 2, is.na), drop = FALSE]
+            temp2 <- temp2[, !apply(temp2, 2, is.na), drop = FALSE]
+            # Merge them together
+            temp <- merge(temp1, temp2, all = TRUE)
+            rownames(temp) <- r
+            # Remove the row from the original rowData
+            rd <- rd[!rownames(rd) %in% r, , drop = FALSE]
+            # Add the data back
+            rd[["rownames_merge_ID"]] <- rownames(rd)
+            temp[["rownames_merge_ID"]] <- rownames(temp)
+            rd <- merge(rd, temp, all = TRUE)
+            rownames(rd) <- rd[["rownames_merge_ID"]]
+            rd[["rownames_merge_ID"]] <- NULL
```

Comment:
When rowData has additional info, it might be that rows with same taxonomy info are not merged together. Here those rows are collapsed together so that there is only one row for each taxa.

Reply:
Quite rare case though...